src/compliance/compliance-requirements.controller.ts

```diff
@@ -137,4 +139,125 @@ export class ComplianceRequirementsController {
     const result = await this.complianceRequirementsService.unpauseRequirements(ticker, params);
     return handleServiceResult(result);
   }
+
+  @ApiOperation({
+    summary: 'Delete single compliance requirement for an Asset',
+    description: 'This endpoint removes referenced compliance requirement for an Asset.',
+  })
+  @ApiParam({
+    name: 'ticker',
+    description:
+      'The ticker of the Asset from which the compliance requirements is to be removed from',
+    type: 'string',
+    example: 'TICKER',
+  })
+  @ApiParam({
+    name: 'id',
+    description: 'The id of the compliance requirement to be deleted',
+    type: 'string',
+    example: '123',
+  })
+  @ApiTransactionResponse({
+    description: 'Details of the transaction',
+    type: TransactionQueueModel,
+  })
+  @ApiTransactionFailedResponse({
+    description: 'The Asset or compliance requirement was not found',
+  })
+  @Delete(':id')
+  public async deleteSingleRequirement(
+    @Param() { id, ticker }: RequirementParamsDto,
+    @Body() params: TransactionBaseDto
+  ): Promise<TransactionResponseModel> {
+    const result = await this.complianceRequirementsService.deleteRequirement(ticker, id, params);
+
+    return handleServiceResult(result);
+  }
+
+  @ApiOperation({
+    summary: 'Delete all compliance requirements for an Asset',
+    description: 'This endpoint removes all compliance requirement for an Asset.',
+  })
+  @ApiParam({
+    name: 'ticker',
+    description:
+      'The ticker of the Asset to which the compliance requirements are to be removed from',
+    type: 'string',
+    example: 'TICKER',
+  })
+  @ApiTransactionResponse({
+    description: 'Details of the transaction',
+    type: TransactionQueueModel,
+  })
+  @ApiTransactionFailedResponse({
+    description: 'The Asset was not found',
+  })
+  @Delete('')
+  public async deleteAllRequirements(
+    @Param() { ticker }: TickerParamsDto,
+    @Body() params: TransactionBaseDto
+  ): Promise<TransactionResponseModel> {
+    const result = await this.complianceRequirementsService.deleteRequirements(ticker, params);
+
+    return handleServiceResult(result);
+  }
+
+  @ApiOperation({
+    summary: 'Add new compliance requirement for an Asset',
+    description: 'This endpoint adds new compliance requirement for an Asset.',
+  })
+  @ApiParam({
+    name: 'ticker',
+    description: 'The ticker of the Asset to which the compliance requirement is to be added to',
+    type: 'string',
+    example: 'TICKER',
+  })
+  @ApiTransactionResponse({
+    description: 'Details of the transaction',
+    type: TransactionQueueModel,
+  })
+  @ApiTransactionFailedResponse({
+    description: 'The Asset was not found',
+  })
+  @Post('')
```

Comment:
The `add` procedure also throws - 
```
   throw new PolymeshError({
      code: ErrorCode.LimitExceeded,
      message: 'Compliance Requirement complexity limit exceeded',
      data: { limit: maxComplexity },
    });
```
Make sure to add this error code in transaction utils and add a exception description for this here

Reply:
I think `422 Unprocessable entity` code should be used